lib/arboreal/instance_methods.rb

```diff
@@ -42,7 +42,8 @@ def root?
 
     # return the root of the tree
     def root
-      root? ? self : ancestors.first
+      return self if root?
+      (root_relation_enabled? and root_ancestor) || ancestors.first
```

Comment:
Both `&&` & `and` work. `(root_relation_enabled? and root_ancestor)` is to be used when value for root_ancestor is set otherwise fallback to ancestors.first